### PR TITLE
chore: change snapshot package

### PR DESCRIPTION
**Pull request by KelvinOm**
/ok-to-test tags="@tag.All"

Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10815637874>
> Commit: 54937a7a64e755959e3420a0a20abcc30ee10d78
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10815637874&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.All`
> Spec:
> <hr>Wed, 11 Sep 2024 17:30:00 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
  - Added support for visual regression testing through new tagging in various test suites.
  - Integrated a new image snapshot plugin to enhance testing capabilities.

- **Bug Fixes**
  - Removed outdated image snapshot assertions to streamline testing processes.

- **Documentation**
  - Updated tags in multiple test suites to improve organization and clarity regarding visual testing.

- **Chores**
  - Upgraded Node.js version in the Cypress Dockerfile for improved performance and compatibility.
  - Simplified Cypress plugin configuration by removing unnecessary functions and enhancing browser compatibility.

### CODE REVIEW

File: app/client/cypress.config.ts [hunk 1]
```diff
@@ -32,6 +33,7 @@ export default defineConfig({
       grepOmitFiltered: true,
     },
     setupNodeEvents(on, config) {
+      addMatchImageSnapshotPlugin(on);
```
Comment (sagar-qa007) [inline app/client/cypress.config.ts:36]: @KelvinOm Just for clarification, do we need this plugin ? if yes then can we remove older plugin?
I have tested another plugin which is forked from original package.

**Review by sagar-qa007 — CHANGES_REQUESTED** [review]

File: app/client/cypress.config.ts (hunk 1 above)
Comment (KelvinOm) [reply app/client/cypress.config.ts:36]: @sagar-qa007 As I wrote in the original post, this plugin fixes a bug with paths. This plugin is also a fork of the original one. I also deleted the old plugin and the places where it was used. Please take a look at the code changes in this PR.

File: app/client/cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilButtonWidgetSnapshot_spec.ts [hunk 2]
```diff
@@ -13,13 +13,15 @@ describe(
     });
 
     it("1. Canvas Mode", () => {
+      cy.viewport(1400, 1200);
```
Comment (jsartisan) [inline app/client/cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilButtonWidgetSnapshot_spec.ts:16]: Do we need to do this for all snapshots?

Comment (KelvinOm) [reply app/client/cypress/e2e/Regression/ClientSide/Anvil/Widgets/AnvilButtonWidgetSnapshot_spec.ts:16]: Opps. I forgot to delete it.

File: app/client/src/constants/PropertyControlConstants.tsx [hunk 3]
```diff
@@ -72,6 +72,7 @@ export interface PropertyPaneControlConfig {
   controlType: ControlType;
   validationMessage?: string;
   dataTreePath?: string;
+  // used to define children property configs when the current property is a section
```
Comment (KelvinOm) [inline app/client/src/constants/PropertyControlConstants.tsx:75]: @jsartisan why is this comment here? How is it related to this PR?

**Review by jsartisan — APPROVED** [review]

**Review by sagar-qa007 — APPROVED** [review]

**Pull request merged**